src/java/org/apache/cassandra/config/Config.java

```diff
@@ -391,6 +391,8 @@ public MemtableOptions()
     // When true, CDC data in SSTable go through commit logs during internodes streaming, e.g. repair
     // When false, it behaves the same as normal streaming.
     public volatile boolean cdc_on_repair_enabled = true;
+
+    public boolean mixed_version_repairs_enabled = true;
```

Comment:
I think generally what I'd want is:

* mixed major version: not allowed
* mixed minor version: allowed

Could we change this to be more oriented around major versions, or do we also have a use case for considering any version difference?

Reply:
Ah i seen on the trunk patch:

> Mixed mode repairs and streaming adds many unknowns and *additional performance impacts during upgrades.*

Reply:
i can go either way, i know id actually like it for minor repairs too but i can be convinced of major repairs only.

Also wondering if we should make it for autorepairs only? currently it blocks manual repairs and autorepairs